src/qml/editorwidgets/RelationEditor/ordered_relation_editor.qml

```diff
@@ -0,0 +1,545 @@
+                if( orderedRelationModel.parentPrimariesAvailable ) {
+                    displayToast( qsTr( 'Adding child feature in layer %1' ).arg( orderedRelationModel.relation.referencingLayer.name ) )
+                    if ( orderedRelationModel.relation.referencingLayer.geometryType() !== QgsWkbTypes.NullGeometry )
+                    {
+                        requestGeometry( relationEditor, orderedRelationModel.relation.referencingLayer );
+                        return;
+                    }
+                    showAddFeaturePopup()
+                }
+                else
+                {
+                    displayToast (qsTr( 'Cannot add child feature: parent primary keys are not available' ), 'warning' )
+                }
+              }
+            }
+          }
+      }
+    }
+
+
+    Component {
+      id: dragDelegate
+
+      MouseArea {
+        id: dragArea
+        pressAndHoldInterval: 130
+
+        property bool held: false
+
+        property int indexFrom: -1
+        property int indexTo: -1
+
+        anchors.left: parent ? parent.left : undefined
+        anchors.right: parent ? parent.right : undefined
+        height: content.height
+
+        drag.target: held ? content : undefined
+        drag.axis: Drag.YAxis
+
+        onPressAndHold: {
+          if (isEnabled) {
+            held = true
+          }
+        }
+        onReleased: {
+           if (held === true) {
+             held = false
+             orderedRelationModel.moveItems(indexFrom, indexTo)
+           } else if (listView.currentIndex !== dragArea.DelegateModel.itemsIndex) {
+             listView.currentIndex = dragArea.DelegateModel.itemsIndex
+             orderedRelationModel.onViewCurrentFeatureChanged(listView.currentIndex)
+           }
+        }
+
+        onClicked: {
+          embeddedPopup.state = isEnabled ? 'Edit' : 'ReadOnly'
+          embeddedPopup.currentLayer = orderedRelationModel.relation.referencingLayer
+          embeddedPopup.linkedRelation = orderedRelationModel.relation
+          embeddedPopup.linkedParentFeature = orderedRelationModel.feature
+          embeddedPopup.feature = model.referencingFeature
+          embeddedPopup.open()
+        }
+
+        Rectangle {
+          id: content
+          anchors {
+              horizontalCenter: parent.horizontalCenter
+              verticalCenter: parent.verticalCenter
+          }
+          width: dragArea.width
+          height: row.implicitHeight + 4
+
+          color: dragArea.held
+                   ? "coral"
+                   : "transparent"
+          Behavior on color { ColorAnimation { duration: 100 } }
+
+          radius: 2
+          Drag.active: dragArea.held
+          Drag.source: dragArea
+          Drag.hotSpot.x: width / 2
+          Drag.hotSpot.y: height / 2
+
+          states: State {
+            when: dragArea.held
+
+            // ParentChange { target: content; parent: root }
```

Comment:
remove?

Reply:
@3nids this was present in you QML, not sure if it is needed?

Reply:
I took this from an example. Not exactly sure what it's doing. Is there any reason to disable it?